core/src/com/unciv/logic/battle/GreatGeneralImplementation.kt

```diff
@@ -0,0 +1,98 @@
+
+        if (unit.hasUnique(UniqueType.GreatGeneralProvidesDoubleCombatBonus, checkCivInfoUniques = true)
+            && greatGeneralModifier.general.isGreatPersonOfType("War")) // apply only on "true" generals
+            return Pair(greatGeneralModifier.general.name, greatGeneralModifier.bonus * 2)
+        return Pair(greatGeneralModifier.general.name, greatGeneralModifier.bonus)
+    }
+
+    /**
+     * Find a tile for accompanying a military unit where the total bonus for all affected units is maximized.
+     *
+     * Used by [SpecificUnitAutomation.automateGreatGeneral].
+     */
+    fun getBestAffectedTroopsTile(general: MapUnit): TileInfo? {
+        // Normally we have only one Unique here. But a mix is not forbidden, so let's try to support mad modders.
+        // (imagine several GreatGeneralAura uniques - +50% at radius 1, +25% at radius 2, +5% at radius 3 - possibly learnable from promotions via buildings or natural wonders?)
+
+        // Map out the uniques sorted by bonus, as later only the best bonus will apply.
+        val generalBonusData = (
+                general.getMatchingUniques(UniqueType.StrengthBonusInRadius).map { GeneralBonusData(general, it) }
+            ).sortedWith(compareByDescending<GeneralBonusData> { it.bonus }.thenBy { it.radius })
+            .toList()
+
+        // Get candidate units to 'follow', coarsely.
+        // The mapUnitFilter of the unique won't apply here but in the ranking of the "Aura" effectiveness.
+        val unitMaxMovement = general.getMaxMovement()
+        val militaryUnitTilesInDistance = general.movement.getDistanceToTiles().asSequence()
+            .map { it.key }
+            .filter { tile ->
+                val militaryUnit = tile.militaryUnit
+                militaryUnit != null && militaryUnit.civInfo == general.civInfo
+                        && (tile.civilianUnit == null || tile.civilianUnit == general)
+                        && militaryUnit.getMaxMovement() <= unitMaxMovement
+                        && !tile.isCityCenter()
+            }
+
+        // rank tiles and find best 
+        val unitBonusRadius = generalBonusData.maxOfOrNull { it.radius }
+            ?: return null
+        return militaryUnitTilesInDistance
+            .maxByOrNull { unitTile ->
+                unitTile.getTilesInDistance(unitBonusRadius).sumOf { auraTile ->
+                    val militaryUnit = auraTile.militaryUnit
+                    if (militaryUnit == null || militaryUnit.civInfo != general.civInfo) 0
+                    else generalBonusData.firstOrNull {
+                        // "Military" as commented above only a small optimization
+                        auraTile.aerialDistanceTo(unitTile) <= it.radius
+                                && (it.filter == "Military" || militaryUnit.matchesFilter(it.filter))
+                    }?.bonus ?: 0
+                }
+            }
+    }
+}
```

Comment:
I'll assume this is unchanged from SomeTroglodyte's implementation, which looked fine last time I looked at this, so I'll skip it for now

Reply:
No, it is changed a bit. See commit `f027696893f2e39019f74c2e3a3001f8de28b2b8` "[Revert of maxGreatGeneralBonusRadius logic](https://github.com/yairm210/Unciv/pull/6818/commits/f027696893f2e39019f74c2e3a3001f8de28b2b8)"